modules/ingest-common/src/main/java/org/elasticsearch/ingest/common/DateFormat.java

```diff
@@ -73,30 +72,22 @@ private long parseMillis(String date) {
     Java {
         @Override
         Function<String, DateTime> getFunction(String format, DateTimeZone timezone, Locale locale) {
-            // in case you are wondering why we do not call 'DateFormatter.forPattern(format)' for all cases here, but only for the
-            // non java time case:
-            // When the joda date formatter parses a date then a year is always set, so that no fallback can be used, like
-            // done in the JodaDateFormatter.withYear() code below
-            // This means that we leave the existing parsing logic in place, but will fall back to the new java date parsing logic, if an
-            // "8" is prepended to the date format string
-            int year = LocalDate.now(ZoneOffset.UTC).getYear();
+
+            // support the 6.x BWC compatible way of parsing java 8 dates
             if (format.startsWith("8")) {
-                DateFormatter formatter = DateFormatter.forPattern(format)
-                    .withLocale(locale)
-                    .withZone(DateUtils.dateTimeZoneToZoneId(timezone));
-                return text -> {
-                    ZonedDateTime defaultZonedDateTime = Instant.EPOCH.atZone(ZoneOffset.UTC).withYear(year);
-                    TemporalAccessor accessor = formatter.parse(text);
-                    long millis = DateFormatters.toZonedDateTime(accessor, defaultZonedDateTime).toInstant().toEpochMilli();
-                    return new DateTime(millis, timezone);
-                };
-            } else {
-                DateFormatter formatter = Joda.forPattern(format)
-                    .withYear(year)
-                    .withZone(DateUtils.dateTimeZoneToZoneId(timezone))
-                    .withLocale(locale);
-                return text -> new DateTime(formatter.parseMillis(text), timezone);
+                format = format.substring(1);
             }
+
+            int year = LocalDate.now(ZoneOffset.UTC).getYear();
+            DateFormatter formatter = DateFormatter.forPattern(format)
+                .withLocale(locale)
+                .withZone(DateUtils.dateTimeZoneToZoneId(timezone));
+            return text -> {
+                ZonedDateTime defaultZonedDateTime = Instant.EPOCH.atZone(ZoneOffset.UTC).withYear(year);
+                TemporalAccessor accessor = formatter.parse(text);
+                long millis = DateFormatters.toZonedDateTime(accessor, defaultZonedDateTime).toInstant().toEpochMilli();
+                return new DateTime(millis, timezone);
```

Comment:
we are still returned a joda time class here, that we need to get rid of. I am happy to fix this in a follow up PR though

Reply:
Thanks @spinscale.  I'd prefer to leave that for a followup PR as I'm sure removing Joda classes will spread out far and wide through the codebase.